### DISCUSSION TITLE
How to set `-R` (`--repo`) correctly?

### DISCUSSION THREAD

**Discussion by csantos1113** [lindell/multi-gitter#308] [Q&A] [answered]
Hey team,

We have a repository with a ton of repositories, and I want to run multi-gitter in some repositories only.

I see there are 2 configs I could use `repo` and `skip-repo`.

This is what I'm trying to do for now:

```
multi-gitter run "node $PWD/read-readme.js" -R MY-ORG/react-ui -m "chore: testing" --dry-run
```

Where `read-readme.js` has:
```
const { readFile } = require("fs").promises;

async function readIt() {
  let data = await readFile("./README.md", "utf8");
  console.log(data);
}
readIt();
```

But the terminal shows that it tries to run in all repos instead:

```
INFO[0009] Running on 189 repositories                  
INFO[0009] Cloning and running script                    repo=MY-ORG/infra
INFO[0011] no data was changed                           repo=MY-ORG/infra
INFO[0011] Cloning and running script                    repo=MY-ORG/algo-testing
INFO[0012] no data was changed                           repo=MY-ORG/algo-testing
INFO[0012] Cloning and running script                    repo=MY-ORG/website
...
...
...
```

### Additional context:
- defining the repo in a `config.yaml` and using it via `--config=./path/to/config.yaml` isn't working for me either.
- I can do `if (process.env.REPOSITORY === 'MY-ORG/react-ui') { // do something }`, but this is not performant because 189 repositories are being cloned even though when I needed only one.
- UPDATE: I have a `~/.multi-gitter/config` with global configuration that will not change
  ```
  token: ghp_myPrivateTokenEtc
  org:
    - MY-ORG
  platform: github
  ```

### Question

- What am I doing wrong?
  - I understood `repo` as "I want to run this script in **only** X,Y,Z repos", Did I misread the docs?
- I haven't tried the `skip-repo` prop because it seems insane that I need to skip/omit 188 repositories

Any guidance would be appreciated! Thank you

**Comment by lindell** [comment, marked as answer]
My first thought that was you defined the organization (-O) option somewhere (like #253), but from your example, that does not seems to be the case. Could it be that you set it in the config file?

**Reply by csantos1113** [reply]
Hey @lindell thanks for the reply. Sorry I missed that part. Yes, I have a `~/.multi-gitter/config` file with the global configuration (things I will never change)

`cat ~/.multi-gitter/config`

```
token: ghp_myPrivateTokenEtc
org:
  - MY-ORG
platform: github
```

**Reply by csantos1113** [reply]
Ok, I just read the #253 issue. And removing the `org` from my global config solved my issue. ✅  Thanks

**Comment by csantos1113** [comment]
Here is how I understood the documentation:

- `org` sets the orgs where you want this to run. all repositories will be included
  - `repo` narrows down the list of repositories where to run this
  - `skip-repo` inverse to `repo`. all repositories in those `orgs` **but** these in skip-repo

But really it is more like:

- `org` sets the orgs where you want this to run. all repositories will be included
- `repo` will be ignored if `org` is set. Use one or the other

(Not sure how to put this in clearer words 😅)

Thanks @lindell for pointing me to the solution
